### PR TITLE
feat: add CMSShape PDF

### PR DESCRIPTION
**Pull request by ikrommyd**
Starting a PR to add the CMSShape PDF into zfit-physics.
Related to https://github.com/zfit/zfit/issues/512

## Proposed Changes

  - Implement CMSShape

## Tests added

  - `tests/test_pdf_cmsshape.py`

## Checklist

 - [x] change approved
 - [x] implementation finished
 - [x] correct namespace imported
 - [x] tests added
 - [x] CHANGELOG updated

### CODE REVIEW

File: zfit_physics/models/pdf_cmsshape.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+from typing import Optional
+
+import numpy as np
+import tensorflow as tf
```
Comment (jonas-eschle) [inline zfit_physics/models/pdf_cmsshape.py:3]: Hm, would be good to have znp. but erf and erfc isn't directly available in the namespace, we should add this in zfit. But TF is fine, it's straight forward to convert

File: zfit_physics/models/pdf_cmsshape.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        gamma: ztyping.ParamTypeInput,
+        m: ztyping.ParamTypeInput,
+        obs: ztyping.ObsTypeInput,
+        *,
+        extended: Optional[ztyping.ExtendedInputType] = None,
+        norm: Optional[ztyping.NormInputType] = None,
+        name: str = "CMSShape",
+    ):
+        params = {"beta": beta, "gamma": gamma, "m": m}
+        super().__init__(obs=obs, params=params, name=name, extended=extended, norm=norm)
```
Comment (jonas-eschle) [inline zfit_physics/models/pdf_cmsshape.py:134]: Looks good, what about the order of the parameters, i.e. `m`? We should follow as close as possible others, i.e. the voigt.

**Review by jonas-eschle — COMMENTED** [review]
Looks good overall, that's the right direction. Docs, (serialization), tests, but you know all that. And as discussed, feel free to add numba-stats as a dependency, that makes sense to test against

File: zfit_physics/models/pdf_cmsshape.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    two = 2.0
+    y = x - m
+    t1 = tf.math.erf(gamma / (two * beta) + beta * y)
+    t2 = tf.math.exp(-((gamma / (two * beta)) ** two) - gamma * y)
+    t3 = tf.math.erfc(-beta * y)
+    return half * (t1 - t2 * t3) + half
+
+
+def cmsshape_integral(limits: ztyping.SpaceType, params: dict, model) -> tf.Tensor:
+    lower, upper = limits.rect_limits
```
Comment (jonas-eschle) [inline zfit_physics/models/pdf_cmsshape.py]:
```suggestion
    lower, upper = limits.limit1d
```
I guess this is the culprit, it has to be of a lower dimension, this should work

File: zfit_physics/models/pdf_cmsshape.py (hunk 2 above)
Comment (ikrommyd) [reply zfit_physics/models/pdf_cmsshape.py:134]: I thought about it. This the the original implementation: https://gitlab.cern.ch/cms-muonPOG/spark_tnp/-/blob/Spark3/RooCMSShape.cc. Here the `alpha` and `peak` are the redundant parameters thet are replaced by just `m`. `m` is a "loc" parameter and is the approcimate center of the distribution. `scipy` uses the name `loc` for such parameters, i.e. in Gaussian `pdf(x, loc=0, scale=1)` but in `zfit` I see `m` being used. We can put it first if you want and the naming convention is also up to debate.

File: zfit_physics/models/pdf_cmsshape.py (hunk 3 above)
Comment (ikrommyd) [reply zfit_physics/models/pdf_cmsshape.py]: It doees indeed fix the problem. But why doesn't it cause a problem here in `relbw`? https://github.com/zfit/zfit-physics/blob/251f4e2de542fcc753dc3ab29c3ae4e01a161815/zfit_physics/models/pdf_relbw.py#L134

File: zfit_physics/models/pdf_cmsshape.py (hunk 1 above)
Comment (ikrommyd) [reply zfit_physics/models/pdf_cmsshape.py:3]: Unless you want to add more stuff into the `znp` namespace we keep it that way for now.

**Comment by ikrommyd** [comment]
@jonas-eschle I'm happy with the current state and testing. Please review it when you got some time.

File: zfit_physics/models/pdf_cmsshape.py (hunk 2 above)
Comment (jonas-eschle) [reply zfit_physics/models/pdf_cmsshape.py:134]: yes, tricky with the names, but as we have m in cauchy, voigt etc, that makes sense to keep it m here too. So m seems reasonable to me. Let's just put it first, so it's consisten with all the other m/mu

File: requirements_dev.txt [hunk 4]
```diff
@@ -2,6 +2,7 @@ bumpversion>=0.5.3
 coverage>=4.5.1
 coverage>=4.5.1
 flake8>=3.5.0
+numba-stats @ git+https://github.com/HDembinski/numba-stats.git
```
Comment (jonas-eschle) [inline requirements_dev.txt]:
```suggestion
numba-stats @ git+https://github.com/HDembinski/numba-stats.git  # CMSShape not yet released (expected 1.8.0)
```

**Review by jonas-eschle — CHANGES_REQUESTED** [review]

**Comment by jonas-eschle** [comment]
Looks good (minus minor things), and you reminded me to also increase the visibility of zfit-physics

File: zfit_physics/models/pdf_cmsshape.py (hunk 2 above)
Comment (ikrommyd) [reply zfit_physics/models/pdf_cmsshape.py:134]: Done in latest commit.

**Comment by ikrommyd** [comment]
@jonas-eschle Alright, let the tests run and merge.

**Review by jonas-eschle — APPROVED** [review]
Looks good! I've just fixed two minor doc things: there was a piece of string laying around in the code and I've converted the docs from MD to RST, which is usually used in Python docstrings (afaik).

**Comment by ikrommyd** [comment]
> Looks good! I've just fixed two minor doc things: there was a piece of string laying around in the code and I've converted the docs from MD to RST, which is usually used in Python docstrings (afaik).

Thanks, I saw md in the relativistic breit wigner implementation so I just followed that. Thanks for removing the string that was lying around, I had forgotten it. Merge it whenever you want!

**Pull request merged**